R/function-filtering.R

```diff
@@ -0,0 +1,147 @@
+#'  matrix representing the biological samples.
+#'
+#' @param y `numeric` For `rowDratio` and `rowBlank`, a numeric matrix
+#'  representing the QC samples and blank samples, respectively.
+#'
+#' @param na.rm `logical(1)` indicate whether missing values (`NA`) should be
+#'  removed prior to the calculations.
+#'
+#' @param mad `logical(1)` indicate whether the *Median Absolute Deviation*
+#'  (MAD) should be used instead of the standard deviation. This is suggested
+#'  for non-gaussian distributed data.
+#'
+#' @return  See individual function description above for details.
+#'
+#' @author Philippine Louail, Johannes Rainer
+#'
+#' @md
+#'
+#' @importFrom stats sd mad median
+#'
+#' @name filteringFunctions
+#'
+#' @references
+#'
+#' Broadhurst D, Goodacre R, Reinke SN, Kuligowski J, Wilson ID, Lewis MR,
+#' Dunn WB. Guidelines and considerations for the use of system suitability
+#' and quality control samples in mass spectrometry assays applied in
+#' untargeted clinical metabolomic studies. Metabolomics. 2018;14(6):72.
+#' doi: 10.1007/s11306-018-1367-3. Epub 2018 May 18. PMID: 29805336;
+#' PMCID: PMC5960010.
+#'
+#' @examples
+#'
+#' ## coefficient of variation
+#' a <- c(4.3, 4.5, 3.6, 5.3)
+#' rsd(a)
+#'
+#' A <- rbind(a, a, a)
+#' rowRsd(A)
+#'
+#' ## Dratio
+#' x <- c(4.3, 4.5, 3.6, 5.3)
+#' X <- rbind(a, a, a)
+#' rowDratio(X, X)
+#'
+#' #' ## Percent Missing
+#' b <- c(1, NA, 3, 4, NA)
+#' percentMissing(b)
+#'
+#' B <- matrix(c(1, 2, 3, NA, 5, 6, 7, 8, 9), nrow = 3)
+#' rowPercentMissing(B)
+#'
+#' ## Blank Rows
+#' test_samples <- matrix(c(13, 21, 3, 4, 5, 6), nrow = 2)
+#' blank_samples <- matrix(c(0, 1, 2, 3, 4, 5), nrow = 2)
+#' rowBlank(test_samples, blank_samples)
+#'
+NULL
+
+#' @export
+#' @rdname filteringFunctions
+#'
+
+rsd <- function(x, na.rm = TRUE, mad = FALSE) {
+    if (mad)
+        mad(x, na.rm = na.rm) / abs(median(x, na.rm = na.rm))
+    else
+        sd(x, na.rm = na.rm) / abs(mean(x, na.rm = na.rm))
+}
+
+#' @rdname filteringFunctions
+#' @export
+rowRsd <- function(x, na.rm = TRUE, mad = FALSE)
+    apply(x, MARGIN = 1, rsd, na.rm = na.rm, mad = mad)
+
+
+#' @export
+#' @rdname filteringFunctions
+#'
+
+rowDratio <- function(x, y, na.rm = TRUE, mad = FALSE){
+    if (mad)
+        vec <- apply(y, 1, mad, na.rm = na.rm) /
+            apply(x, 1, mad, na.rm = na.rm)
+    else
+        vec <- apply(y, 1, sd, na.rm = na.rm) /
+        apply(x, 1, sd, na.rm = na.rm)
+}
+
+
+#' @export
+#' @rdname filteringFunctions
+
+percentMissing <- function(x){
+   ((sum(is.na(x))) / length(x))*100
+}
+
+#' @export
+#' @rdname filteringFunctions
+#'
+rowPercentMissing <- function(x){
+    apply(x, MARGIN = 1, percentMissing)
+}
+
+
+#' @export
+#' @rdname filteringFunctions
+#'
+
+rowBlank <- function(x, y, na.rm = TRUE){
+    m_samples <- apply(x, 1, mean, na.rm = na.rm)
+    m_blank <- apply(y, 1, mean, na.rm = na.rm)
+    vec <- m_samples > 2 * m_blank
```

Comment:
here it's enough to have

```r
m_samples > 2 * m_blank
```

otherwise (with `vec <-`) the function will not return anything (I believe?). 

Also, from the name of the function I would have expected that the function returns `TRUE` if the feature represents signal also found in blanks - i.e. the opposite what is returned now? So, basically, `rowBlank` returns `TRUE` if the signal in test samples is not at least 2 times higher than the signal in blanks, or in other words, if the feature was also found in blank samples.

Reply:
Yes i realised that while writing the vignette, I am now reporting TRUE for potential contaminants, it makes more sense

Reply:
- just waiting to make all the changes before pushing :)

Reply:
the reason to drop the final `vec <- ` is also that R does not need to initialize another (then unneeded) variable and copy content in it.